custom-graders/v2/DemoPythonGrader/README.md

```diff
@@ -0,0 +1,36 @@
+# Demo Python Custom Grader for Coursera
+This is a minimal autograder template for graded Programming Assignments on the Coursera platform.
+<br><br>Graded technical assessments create immense value for our learners, and Python is one of the most popular languages in the industry as well as on the Coursera platform.
+<br><br>This template was created to introduce our partners to the minimum requirements for implementing auto-grading on Coursera. It is meant for partners to use as a starting point.
+
+# Grader workflow
+At a high-level, this custom grader accomplishes autograding of student submissions following these steps:
+1. The Dockerfile creates the image. It sets up a Python 3 environment and installs dependencies. Then, it copies over the relevant files for grading, and assigns appropriate permissions. It launches into the `ENTRYPOINT`, which in this grader is defined as `grader.py`.
+2. In `grader.py`, the `main` function assigns number of test cases to run against the submission & solution.
+3. The grader then checks for the student submission in the `/shared/submission/` directory. This is the standard submission directory across all courses on Coursera, and is a read-only directory.
+4. After checking that the submission has a `.py` extension, test cases are randomly generated according to the number of test cases defined in Step 2. Corresponding solutions are also generated using the files in `/autograder/solutions/`.
+5. The grader then checks for a matching partId and runs the test cases through the learner submission.
+6. After comparing the learner submission outputs to the solutions, the grader calculates the total score and writes to `/shared/feedback.json` with `fractionalScore` and `feedback` fields. On the Coursera platform, this feedback can be viewed by the learner; however, the learner does not have access to `stderr` or other container logs (only course admin and staff do). It is important to include appropriate feedback from the custom grader into the `feedback` field of this JSON object.
```

Comment:
how about if we phrase this way?
..the learner does not have access to container logs (contains both stdout and stderr logs).

Reply:
Great catch! I'll edit that now and merge 👍